### ISSUE TITLE
Potential Timezone Bug

### ISSUE THREAD

**Issue by jasontt33** [dotCMS/core#20013]
**Describe the bug**
dotcms not handling Time Zone CST properly or Postgres no longer accepts CST as a valid timezone as it could mean a timezone in China or the Americas

**To Reproduce**
1. log into dotcms and go to Configuration
2. Select CST as timezone and click save
3. postgres will eventually choke so watch the pg_logs or dotcms log will tell ya too

**Expected behavior**
backend becomes unresponsive and eventually site degrades
This bug brought down two customers because of this. 
I suspect the values / params we send to the backend of dotcms (db) need to be updated

**Screenshots**
n/a

**Desktop (please complete the following information):**
n/a 
**Smartphone (please complete the following information):**
n/a

**Additional context**
See Jason Thompson in Engineering if you need more details.

**Comment by john-thomas-dotcms** [comment]
The core issue here is that we present time zones to the customer that are invalid.  So, if a customer changes their time zone config in the BE, it can break dotCMS.

So, we need to remove the invalid time zones from the `Basic Config` tab in the `Configuration` portlet.

There is some disagreement or misunderstanding about where that list of time zones is coming from.  It was believed that it was being pulled from the OS, but when tested, he found that even when the OS was fully updated with the latest time zones (which excluded the invalid time zones), the invalid time zones were still displayed as options in dotCMS.

**Comment by wezell** [comment]
The list of timezones is coming from java.  This is not a super easy change - what broke here was the database timezone -  `CST` might be a valid timezone in MySQL, Oracle or MSSQL and `CST6CDT` might not be.

In messing around here we might break customers who are currently successfully running some of these "invalid" timezones.

The fix was to change the timezone to the right value. Maybe when a customer changes the value here, we just have an alert that tells the customer if they have selected a timezone that is not compatible with their database.

**Comment by wezell** [comment]
So, when we save the company, we need to run a sql query that tries to use this timezone in the ways that broke the above instance and if that query fails, we throw a GROWL ERROR that says that `The selected timezone does not work with $DATABASE, please select another.`

**Comment by swicken-dotcms** [comment]
`[25/02/21 11:29:32:359 EST] ERROR reindex.ReindexThread: ReindexThread Exception
com.dotmarketing.exception.DotDataException: java.sql.SQLTransientConnectionException: jdbc/dotCMSPool - Connection is not available, request timed out after 3000ms.{
  "SQL": ["select cluster_server.*, sitelic.id as license_id, sitelic.lastping as lastping from cluster_server left join sitelic on server_id = serverid   where sitelic.id is not null and lastping > ? order by server_id"],
  "maxRows": [-1],
  "offest": [0],
  "params": "2021-02-25 10:24:29.358"
}
        at com.dotmarketing.common.db.DotConnect.loadResult(DotConnect.java:273) ~[dotcms_5.3.8.2_01deb9e.jar:?]
        at com.dotmarketing.common.db.DotConnect.loadObjectResults(DotConnect.java:925) ~[dotcms_5.3.8.2_01deb9e.jar:?]
        at com.dotcms.enterprise.cluster.ServerFactoryImpl.getReindexingServers(SourceFile:166) ~[ee-5.3.8.jar:?]
        at com.dotcms.cluster.business.ServerAPIImpl.ge`

**Comment by victoralfaro-dotcms** [comment]
Logs from postgres:
```
$ sudo tail -f data/log/postgresql-2021-03-12_160350.log                                                                                                                                           2021-03-12 16:03:50.397 CST [82880] LOG:  database system was shut down at 2021-03-12 16:03:37 CST
2021-03-12 16:03:50.407 CST [82878] LOG:  database system is ready to accept connections
2021-03-12 16:06:36.885 CST [85341] FATAL:  invalid value for parameter "TimeZone": "CST"
2021-03-12 16:06:37.169 CST [85342] FATAL:  invalid value for parameter "TimeZone": "CST"
2021-03-12 16:06:37.553 CST [85343] FATAL:  invalid value for parameter "TimeZone": "CST"
2021-03-12 16:06:38.124 CST [85344] FATAL:  invalid value for parameter "TimeZone": "CST"
2021-03-12 16:06:38.975 CST [85346] FATAL:  invalid value for parameter "TimeZone": "CST"
2021-03-12 16:06:40.244 CST [85347] FATAL:  invalid value for parameter "TimeZone": "CST"
2021-03-12 16:06:42.151 CST [85349] FATAL:  invalid value for parameter "TimeZone": "CST"
2021-03-12 16:06:44.999 CST [85350] FATAL:  invalid value for parameter "TimeZone": "CST"
2021-03-12 16:06:48.007 CST [85351] FATAL:  invalid value for parameter "TimeZone": "CST"
```

**Comment by wezell** [comment]
Did you test saving the company from the configuration portlet?

**Comment by dsilvam** [comment]
Needs work: 
1. When trying to set _Central Standard Time (CST)_ it pops up a modal saying _Invalid Timezone CST_, which is ok, but I think the previous option should stay selected. 
2. The full stacktrace is getting logged. I think we should only log the error message.

**Comment by dsilvam** [comment]
Passed Internal QA: A popup is shown when attempted to save an invalid timezone. Previous value remains selected and only one-line ERROR is logged.

**Comment by bryanboza** [comment]
Fixed, tested on release 21.04 // Postgres // FF
